api/refresh_token_logic.py

```diff
@@ -0,0 +1,83 @@
+from flask import request
+from user_agents import parse
+import requests
+import hashlib
+import time
+import os
+from typing import Optional
+
+class User_info_service:
+    def __init__(self, accass_token_dal, refresh_token_dal):
+        self._access_token_dal = accass_token_dal
+        self._refresh_token_dal = refresh_token_dal
+    def get_user_device(self) -> str:
```

Comment:
I bet PyCharm treats it as a warning (no empty lines before method). If PyCharm is silent about this (because you disabled the warnings - that's the only way) - please follow the coding standards for Python manually